### PR TITLE
Save funds does not use execution delegate

### PR DESCRIPTION
**Pull request by hugoroussel**
Addresses [https://github.com/cantinasec/review-fantasy/issues/20](https://github.com/cantinasec/review-fantasy/issues/20)

### CODE REVIEW

File: src/Minter.sol [hunk 1]
```diff
@@ -679,6 +680,14 @@ contract Minter is IMinter, AccessControlDefaultAdminRules, ReentrancyGuard, Lin
      * @param amount Transfer amount.
      */
     function saveFunds(address paymentToken, address to, uint256 amount) public onlyRole(DEFAULT_ADMIN_ROLE) {
-        _transferTo(paymentToken, address(this), to, amount);
+        if (amount == 0) {
```
Comment (bane-the-dev) [inline src/Minter.sol]: why no make this a require and revert if amount is 0?

File: src/Minter.sol [hunk 2]
```diff
@@ -679,6 +680,14 @@ contract Minter is IMinter, AccessControlDefaultAdminRules, ReentrancyGuard, Lin
      * @param amount Transfer amount.
      */
     function saveFunds(address paymentToken, address to, uint256 amount) public onlyRole(DEFAULT_ADMIN_ROLE) {
```
Comment (bane-the-dev) [inline src/Minter.sol:682]: Also we should have a test for this function

**Review by bane-the-dev — CHANGES_REQUESTED** [review]

File: src/Minter.sol [hunk 3]
```diff
@@ -14,6 +14,7 @@ pragma solidity ^0.8.20;
 import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
 import "@openzeppelin/contracts/access/extensions/AccessControlDefaultAdminRules.sol";
 import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
```
Comment (bane-the-dev) [inline src/Minter.sol:17]: Should import IERC20 instead

**Review by bane-the-dev — CHANGES_REQUESTED** [review]

File: src/Minter.sol (hunk 1 above)
Comment (hugoroussel) [reply src/Minter.sol]: I would just fully remove the check

File: src/Minter.sol (hunk 3 above)
Comment (hugoroussel) [reply src/Minter.sol:17]: fixed

File: src/Minter.sol (hunk 2 above)
Comment (hugoroussel) [reply src/Minter.sol:682]: fixed

**Pull request merged**